cephprimarystorage/cephprimarystorage/cephagent.py

```
@@ -1399,21 +1399,27 @@ def get_volume_backing_chain(self, req):
+                    shell.run('rbd snap unprotect %s' % path)
+                    shell.call('rbd snap rm %s' % path)
+                    rsp.undeletedInstallPaths[rsp.undeletedInstallPaths.index(installPath)] = installPath.split("@")[0]
+                else:
+                    vol_path = path
+                self._ensure_existing_volume_has_no_snapshot(vol_path)
+                watchers = self._get_watcher(vol_path)
+                if watchers:
+                    raise Exception('volume %s has watchers %s, can not delete it' % (vol_path, watchers))
+                driver.do_deletion(cmd, vol_path)
+                rsp.undeletedInstallPaths.remove(installPath.split("@")[0])
+        except Exception:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)